src/main/java/com/aliyun/odps/jdbc/OdpsResultSet.java

```diff
@@ -629,31 +541,7 @@ public SQLXML getSQLXML(String columnLabel) throws SQLException {
 
   @Override
   public short getShort(int columnIndex) throws SQLException {
-    Object obj = getObject(columnIndex);
-    if (obj == null) {
-      return 0;
-    } else if (obj instanceof Long) {
-      return ((Long) obj).shortValue();
-    } else if (obj instanceof Double) {
-      return ((Double) obj).shortValue();
-    } else if (obj instanceof BigDecimal) {
-      return ((BigDecimal) obj).shortValue();
-    }
-
-    String strVal;
-    if (obj instanceof String) {
-      strVal = (String) obj;
-    } else if (obj instanceof byte[]) {
-      strVal = new String((byte[]) obj);
-    } else {
-      throw new SQLException(
-          "Illegal cast column " + columnIndex + " to short: " + obj.toString());
-    }
-    try {
-      return Long.valueOf(strVal).shortValue();
-    } catch (java.lang.NumberFormatException e) {
-      throw new SQLException("can not parse bigint: " + strVal, e);
-    }
+    return (short) getLong(columnIndex);
```

Comment:
@lyman   
just like we could implement getBytes() based on getString()
we could implement getShort()/getInt()/getByte() based on getLong()